### PR TITLE
feat(plan): compact context and start a fresh session when a plan is approved

### PR DESCRIPTION
**Pull request by edenreich**
## Summary

Closes #677.

When a plan is approved in Plan Mode, the agent now **compacts the (usually exploration-heavy) planning conversation into a summary, starts a fresh session, and re-reads the approved plan from disk to execute** — so execution begins with an optimized, much smaller context window.

Planning is read-heavy (`Read`/`Grep`/`Tree` all over the codebase), and none of that exploration is needed once the plan is accepted. This reuses the existing `/compact` machinery at plan-approval time.

## Behaviour

| `compact.enabled` | Auto-compact at `auto_at` (80%) | Compact on plan approval |
| --- | --- | --- |
| `true` (default) | ✅ | ✅ |
| `false` | ❌ | ✅ (always) |

Compaction on plan approval runs **regardless of `compact.enabled`** — that flag only governs the automatic mid-conversation compaction and the headless session rollover. It also **fails open**: if summarization errors or times out, execution still proceeds on the un-compacted conversation.

> Note: the approved plan is never lost — it lives on disk under `<configDir>/plans/…` and is captured in the summary; the continuation prompt points the agent back at that file to execute step by step.

## Changes

- **Plan-path plumbing** — the saved plan path from the `RequestPlanApproval` tool result flows through `PlanApprovalRequestedEvent` → `PlanApprovalUIState`, so the post-approval continuation prompt can reference it (`internal/agent/agent.go`, `internal/domain/{events,state,interfaces}.go`, `internal/services/state_manager.go`, `internal/services/approvalcoord/coordinator.go`).
- **Orchestration** — `HandlePlanApprovalResponseEvent` captures the path and, on accept, runs a new `compactThenExecutePlanCmd`: async optimize under a 70s timeout → reseed a new session → add a plan-path-aware continue message → resume the agent (`internal/handlers/chat_handler.go`, new `internal/handlers/chat_compaction.go`).
- **Shared helpers** — extracted `optimizeWithTimeout` / `reseedConversationWithMessages` / `nonHiddenMessages`; the `/compact` shortcut now reuses them (behaviour unchanged).
- **Optimizer always constructed** — `internal/container/container.go` builds the `ConversationOptimizer` unconditionally so approval compaction works even with `compact.enabled: false` (`force=true` bypasses the enabled check).
- **Docs** — new "Context compaction on approval" section in `docs/plan-mode.md`; fixed the stale `compact.enabled` default (it is `true`, not `false`) in `docs/configuration-reference.md`.

## Testing

- `task build`, full `go test ./...`, `golangci-lint` (0 issues), `markdownlint` — all green.
- New unit tests: `extractPlanPath` (edge cases), `checkToolResultsStatus` path extraction, `planExecutionContinuePrompt`, and `compactPlanningContext` (reseed / no-reduce / absent optimizer / no model), plus plan-path forwarding in the coordinator test.

### Manual check to run before merge
Interactive TUI + live gateway (not runnable in CI): Shift+Tab into Plan Mode, ask for a multi-file change so the model reads several files, **Accept** the plan, and confirm the history collapses to a `--- Context Summary ---`, the agent re-`Read`s the plan file, and `/context` shows a smaller token count. Repeat with **Accept & Auto-Approve**, and with `compact.enabled: false`.

## Docs follow-up

A matching `[DOCS]` issue should be filed for `inference-gateway/docs` (Plan Mode / compaction pages) per the docs follow-up rule.

### CODE REVIEW

File: internal/container/container.go [hunk 1]
```diff
@@ -314,19 +314,24 @@ func (c *ServiceContainer) initializeDomainServices() {
...
-			Client:            summaryClient,
-			Config:            c.config,
-			Tokenizer:         c.tokenizer,
-			Repo:              c.conversationRepo,
-		})
+	// The optimizer is constructed unconditionally: compact.enabled only gates the
+	// automatic mid-conversation compaction (OptimizeMessages with force=false) and
+	// the headless session rollover below. Compaction on plan approval always runs
+	// (it calls OptimizeMessages with force=true, which ignores the enabled flag),
+	// so the optimizer must exist even when compact.enabled is false.
```
Comment (edenreich) [inline internal/container/container.go]:
```suggestion
```

File: internal/handlers/chat_handler.go [hunk 2]
```diff
@@ -466,10 +466,26 @@ func (h *ChatHandler) HandleUserQuestionRequestedEvent(
 func (h *ChatHandler) HandlePlanApprovalResponseEvent(
 	msg domain.PlanApprovalResponseEvent,
 ) tea.Cmd {
+	// Capture the saved plan path before the coordinator clears the UI state; on
+	// approval we compact the planning context away and point the agent back at
+	// the plan file to execute.
```
Comment (edenreich) [inline internal/handlers/chat_handler.go]:
```suggestion
```

File: internal/handlers/chat_handler.go [hunk 3]
```diff
@@ -466,10 +466,26 @@ func (h *ChatHandler) HandleUserQuestionRequestedEvent(
...
+
 	cmd, restart := h.approvalCoordinator.HandlePlanApprovalResponse(msg)
 	if !restart {
 		return cmd
 	}
+
+	// Compaction on approval runs independently of compact.enabled (that flag only
+	// governs the automatic mid-conversation compaction). We only need the optimizer
+	// to be constructed and a saved plan file to point the agent back at; otherwise
+	// fall back to executing in place.
```
Comment (edenreich) [inline internal/handlers/chat_handler.go]:
```suggestion
```
